### ISSUE TITLE
possible problem with subset(argo)

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1138]
**1.**

*Background.* I am not sure that this is a problem but I'm posting it anyway, for the record.  I've been making oce handle several more BioArgo names. This work is still in progress, and **not** pushed to GH; I am unlikely to push until the end of the day, so I request that other developers stay away from `R/argo.R` until I fix this.

Anyway, what I'm finding is that if I reconstruct `data(argo)` then a build test fails. I think this may be a problem in (a) the regular expressions that are used to translate names (since this is what I've been working on) or some ripple that is causing problems in `subset,argo-method` (possibly near line 330, but this is yet to be confirmed, and I'm noting it here as a diary item).

The main reason I'm posting this is that it seems likely that I will be altering more than just the name-translation code, and also that I might be altering up to three lines added by CR on 2016-04-04.

Below are the lines that fail in the build test ... again, this is for my notes; nobody else need consider this comment.

```R
expect_equal(a@metadata$flags$salinityQc, argo@metadata$flags$salinityAdjustedQc) 
expect_equal(a@metadata$flags$pressure, argo@metadata$flags$pressureAdjusted) 
```

**2.**

With newly-constructed `data(argo)` and with `a` created with `a <- subset(argo, "adjusted")`:

```R
> str(argo@metadata$flags)
List of 6
 $ pressureQC           : num [1:56, 1:207] 1 1 1 1 1 1 1 1 1 1 ...
 $ pressureAdjustedQC   : num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
 $ salinityQC           : num [1:56, 1:207] 1 1 1 1 1 1 1 1 1 1 ...
 $ salinityAdjustedQC   : num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
 $ temperatureQC        : num [1:56, 1:207] 1 1 1 1 1 1 1 1 1 1 ...
 $ temperatureAdjustedQC: num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
> str(a@metadata$flags)
List of 6
 $ pressureQC           : num [1:56, 1:207] 1 1 1 1 1 1 1 1 1 1 ...
 $ pressureAdjustedQC   : num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
 $ salinityQC           : num [1:56, 1:207] 1 1 1 1 1 1 1 1 1 1 ...
 $ salinityAdjustedQC   : num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
 $ temperatureQC        : num [1:56, 1:207] 1 1 1 1 1 1 1 1 1 1 ...
 $ temperatureAdjustedQC: num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
> 
```

**3.**

With old `data(argo)`
```R
> str(argo@metadata$flags)
List of 6
 $ pressure           : num [1:56, 1:207] 1 1 1 1 1 1 1 1 1 1 ...
 $ pressureAdjusted   : num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
 $ salinity           : num [1:56, 1:207] 1 1 1 1 1 1 1 1 1 1 ...
 $ salinityAdjusted   : num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
 $ temperature        : num [1:56, 1:207] 1 1 1 1 1 1 1 1 1 1 ...
 $ temperatureAdjusted: num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
> str(a@metadata$flags)
List of 3
 $ pressure   : num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
 $ salinity   : num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
 $ temperature: num [1:56, 1:207] 2 2 2 2 2 2 2 2 2 2 ...
```

**4.**

Oh, I see. The `QC` suffix is not being handled properly in the new name-translation code.

**Comment by dankelley** [comment]
Working now in my non-pushed version. Will close issue late today after more testing.

**Comment by dankelley** [comment]
Fixed in `develop` commit c74ab47b6e54e416ffca17d3119cd7635df17b09
